### PR TITLE
safer temp names

### PR DESCRIPTION
**Pull request by ocefpaf**
~~Not ready for review yet.~~

This is ready for review now.

### CODE REVIEW

File: conda_lock.py [hunk 1]
```diff
@@ -17,19 +17,28 @@
...
-FAKE_PKGS_ROOT = "/tmp/something/that/does/not/exist"
 
 
+def _safe_fake_prefix_name():
+    args = ("conda", "env", "list", "--quiet", "--json")
+    json_output = subprocess.check_output(args)
+    envs = json.loads(json_output)["envs"]
+    PREFIX_NAMES = [pathlib.Path(env).stem for env in envs if "envs" in env]
+
+    while (FAKE_PREFIX_NAME := next(tempfile._get_candidate_names())) not in PREFIX_NAMES:
```
Comment (ocefpaf) [inline conda_lock.py]: I don't like this b/c it will limit to Pyhton 3.8 and uses a private method.

Comment (mariusvniekerk) [reply conda_lock.py]: basically we just need a way to get a thing to find a writable prefix (which is somehow required for conda to solve even in dry-run mode)

Comment (mariusvniekerk) [reply conda_lock.py]: So we can just make a new tempdir and pick a random prefix inside there

File: conda_lock.py [hunk 2]
```diff
@@ -16,31 +16,33 @@
...
-    json_output = subprocess.check_output(args, env=env)
-    # print(json_output)
+    with tempfile.TemporaryDirectory() as CONDA_PKGS_DIRS:
+        env.update({"CONDA_SUBDIR": platform, "CONDA_PKGS_DIRS": CONDA_PKGS_DIRS})
+
+        args = [
+            "conda",
+            "create",
+            "--prefix",
+            f"{CONDA_PKGS_DIRS}_prefix",
```
Comment (ocefpaf) [inline conda_lock.py]: @mariusvniekerk, after asking around I believe this is safer b/c, if one day dry-run does the right thing and fails with an existing name, we would have a broken conda-lock.

File: conda_lock.py [hunk 3]
```diff
@@ -16,31 +16,33 @@
...
-        "--override-channels",
-        "--dry-run",
-        "--json",
-    ]
-    for channel in channels:
-        args.extend(["-c", channel])
-    args.extend(specs)
 
     env = dict(os.environ)
-    env.update({"CONDA_SUBDIR": platform, "CONDA_PKGS_DIRS": FAKE_PKGS_ROOT})
```
Comment (ocefpaf) [inline conda_lock.py]: Changed the names of the variables to match the conda equivalent for readability. I also expanded all the short optional args for the same reason.

File: conda_lock.py (hunk 2 above)
Comment (mariusvniekerk) [reply conda_lock.py]:
```suggestion
            f"{CONDA_PKGS_DIRS}/prefix",
```

Comment (mariusvniekerk) [reply conda_lock.py]: or maybe with pathlib's join

**Pull request merged**